pkg/operator/controller/controller.go

```diff
@@ -169,8 +169,10 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				}
 			} else {
 				// Handle everything else.
+				if err := r.ensureClusterIngressFinalized(ingress); err != nil {
+					errs = append(errs, fmt.Errorf("failed to finalize clusteringress %s/%s: %v", ingress.Namespace, ingress.Name, err))
+				}
 				if err := r.ensureClusterIngress(ingress, dnsConfig, infraConfig); err != nil {
```

Comment:
Because this isn't in an `else` of the finalizer enforcement, does that leave a gap where we can fail to set the finalizer, reconcile the ingress anyway, get deleted, and then fall through to deletion handling without a finalizer? Can we actually reconcile before we've successfully enforced the finalizer invariant?

Reply:
Agree, this should be made an `else if` clause to the preceding `if`.

Reply:
Thanks for helping me better understand the logic. I'll update.